main.go

```diff
@@ -97,15 +101,81 @@ func setThreads() {
 	}
 }
 
+func unixAddr(uri string) (string, error) {
+	if uri == "" {
+		return "", errors.New("empty uri")
+
+	}
+	addr, err := url.Parse(uri)
+	if err != nil {
+		return "", err
+	}
+	if addr.Scheme != "" && addr.Scheme != "unix" {
+		return "", errors.New("invalid address scheme")
+	}
+	return addr.Host + addr.Path, nil
+}
+
+func printAgentLogs(sock string) error {
+	// Don't return an error if nothing has been provided.
+	// This flag is optional.
```

Comment:
'not' optional? the comment does not seem to align with how the code reads?

Reply:
you're right